tmt/steps/provision/__init__.py

```diff
@@ -1158,10 +1164,14 @@ def go(self) -> None:
 
                     if self.is_multihost:
                         self.info('')
+                except (tmt.utils.RunError, tmt.utils.ProvisionError) as error:
+                    self.fail(str(error))
+                    raise
                 finally:
                     if isinstance(phase, ProvisionPlugin):
                         guest = phase.guest()
-                        if guest:
+                        assert guest is not None
```

Comment:
There is a flow (internal `minute --list-images`) where guest is None -> raises assert. 
Typing says we allow guest to be None (`-> Optional[Guest]`) so code should cope with none guest.

Reply:
I'd say `if guest and guest.is_ready or self.opt('dry'):` should be OK

Reply:
Please remove the assert as well.

Reply:
Removed.